packages/contracts/src/PoolMetadataRegistry.sol

```diff
@@ -12,15 +16,21 @@ interface IPoolMetadataRegistry {
 /// @title A Pool Metadata dApp
 /// @author Bleu LLC
 /// @notice This contract provides a registry for Balancer's Pools metadata
-contract PoolMetadataRegistry is IPoolMetadataRegistry {
+contract PoolMetadataRegistry is IPoolMetadataRegistry, PoolMetadataAuthorization {
     IVault private immutable _vault;
-
+    bytes32 private _poolId;
     mapping(bytes32 => string) public poolIdMetadataCIDMap;
 
-    constructor(IVault vault) {
+    constructor(IVault vault) Authentication(bytes32(uint256(address(this)))) PoolMetadataAuthorization() {
         _vault = vault;
     }
 
+    modifier onlyRegisteredPool(bytes32 poolId) {
+        _poolId = poolId;
```

Comment:
Any reason to set this value here? I wouldn't expect this function to have side effects in the instance variables.

Reply:
```js
function setPoolMetadata(bytes32 poolId, string memory metadataCID)
        public
        onlyRegisteredPool(poolId)
        authenticate
    {
        poolIdMetadataCIDMap[poolId] = metadataCID;
        emit PoolMetadataUpdated(poolId, metadataCID);
    }
```
as I said above to @mendesfabio, we need the poolId inside the getOwner (a function that does not receive parameters)

1. The authenticate modifier calls the `canPerform`.
2. the `canPerform` receive the `bytes32 actionId` and the `address account`
3. the `canPerform` calls `getOwner` without any arguments

As we need the poolId, I created a variable and we set its value before performing the authenticate modifier on the setPoolMetadata.